code/game/objects/objs.dm

```diff
@@ -208,22 +208,22 @@
 
 /obj/set_dir(ndir)
 	. = ..()
-	if(directional_offset)
+	if(directional_offset && anchored)
 		update_directional_offset()
 
 /obj/Move()
 	. = ..()
-	if(directional_offset)
+	if(directional_offset && anchored)
```

Comment:
Yeah, that's a brainfart on my part. I'll remove that line in Move().

And, if I null the offsets when not anchored, it means anything else that might set the default offset would get overwritten when unanchored. So I'd rather take a more prudent approach to it and let whatever unanchor the thing handle its new default offset. That's especially tricky for items, since they tend to mess with pixel offsets a bunch.